central/imagev2/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,646 @@
+	s.truncateTable(postgresSchema.ImageCvesV2TableName)
+}
+
+func (s *ImageV2DataStoreTestSuite) TestSearch() {
+	image := getTestImageV2("img1")
+
+	ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowFixedScopes(
+		sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+		sac.ResourceScopeKeys(resources.Image),
+	))
+
+	// Upsert image.
+	s.NoError(s.datastore.UpsertImage(ctx, image))
+
+	// Basic unscoped search.
+	results, err := s.datastore.Search(ctx, pkgSearch.EmptyQuery())
+	s.NoError(err)
+	s.Len(results, 1)
+
+	q := pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.ImageID, image.GetId()).ProtoQuery()
+	results, err = s.datastore.Search(ctx, q)
+	s.NoError(err)
+	s.Len(results, 1)
+
+	q = pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.ImageSHA, image.GetSha()).ProtoQuery()
+	results, err = s.datastore.Search(ctx, q)
+	s.NoError(err)
+	s.Len(results, 1)
+
+	// Sort by impact score
+	q = pkgSearch.EmptyQuery()
+	q.Pagination = &v1.QueryPagination{
+		SortOptions: []*v1.QuerySortOption{
+			{
+				Field: pkgSearch.ImpactScore.String(),
+			},
+		},
+	}
+	results, err = s.cveDataStore.Search(ctx, q)
+	s.NoError(err)
+	s.Equal([]string{"cve2", "cve1"}, splitFlattenedIDs(pkgSearch.ResultsToIDs(results)))
+
+	// Sort by impact score: reversed
+	q = pkgSearch.EmptyQuery()
+	q.Pagination = &v1.QueryPagination{
+		SortOptions: []*v1.QuerySortOption{
+			{
+				Field:    pkgSearch.ImpactScore.String(),
+				Reversed: true,
+			},
+		},
+	}
+	results, err = s.cveDataStore.Search(ctx, q)
+	s.NoError(err)
+	s.Equal([]string{"cve1", "cve2"}, splitFlattenedIDs(pkgSearch.ResultsToIDs(results)))
+
+	// Upsert new image.
+	newImage := getTestImageV2("img2")
+	newImage.GetScan().Components = append(newImage.GetScan().GetComponents(), &storage.EmbeddedImageScanComponent{
+		Name:    "comp3",
+		Version: "ver1",
+		Vulns: []*storage.EmbeddedVulnerability{
+			{
+				Cve:               "cve3",
+				VulnerabilityType: storage.EmbeddedVulnerability_IMAGE_VULNERABILITY,
+			},
+		},
+	})
+	s.NoError(s.datastore.UpsertImage(ctx, newImage))
+
+	// Search multiple images.
+	images, err := s.datastore.SearchRawImages(ctx, pkgSearch.EmptyQuery())
+	s.NoError(err)
+	s.Len(images, 2)
+
+	// Search for just one image.
+	q = pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.ImageID, image.GetId()).ProtoQuery()
+	images, err = s.datastore.SearchRawImages(ctx, q)
+	s.NoError(err)
+	s.Len(images, 1)
+
+	// Search by CVE.
+	q = pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.CVE, "cve1").ProtoQuery()
+	images, err = s.datastore.SearchRawImages(ctx, q)
+	s.NoError(err)
+	s.Len(images, 2)
+
+	q = pkgSearch.NewQueryBuilder().AddExactMatches(pkgSearch.CVE, "cve3").ProtoQuery()
+	results, err = s.datastore.Search(ctx, q)
+	s.NoError(err)
+	s.Len(results, 1)
+	s.Equal(newImage.GetId(), results[0].ID)
+
+	// TODO(ROX-29941) : Uncomment below tests after scoping is implemented at search framework level
```

Comment:
The scope should be added.  If I missed a query path, let me know.

Reply:
Worked pretty well. The tests passed after I uncommented them